aspen/http/resource.py

```diff
@@ -7,25 +8,18 @@ class Static(object):
     """
 
     def __init__(self, request_processor, fspath, raw, media_type):
-        self.request_processor = request_processor
+        assert type(raw) is bytes  # sanity check
         self.raw = raw
         self.media_type = media_type
         if media_type == 'application/json':
-            self.media_type = self.request_processor.media_type_json
+            self.media_type = request_processor.media_type_json
+        if self.media_type.startswith('text/'):  # XXX this is bad
```

Comment:
Remember that `charset_static` is a configuration knob, so the app developer has control, and its default value is `None`, so it doesn't cost anything unless it's explicitly turned on.